packages/sdk/typescript/human-protocol-sdk/test/kvstore.test.ts

```diff
@@ -0,0 +1,246 @@
+      );
+      expect(mockKVStoreContract.setBulk).toHaveBeenCalledWith(
+        ['key1', 'key2'],
+        ['value1', 'value2']
+      );
+    });
+
+    test('should throw an error when attempting to set a value without a signer', async () => {
+      const getClientParamsMock = InitClient.getParams as jest.Mock;
+      getClientParamsMock.mockResolvedValue({
+        provider,
+        network,
+      });
+      kvStoreClient = new KVStoreClient(await InitClient.getParams(provider));
+
+      await expect(
+        kvStoreClient.setBulk(['key1', 'key2'], ['value1', 'value2'])
+      ).rejects.toThrow(
+        Error(
+          'Failed to set bulk values: sending a transaction requires a signer (operation="sendTransaction", code=UNSUPPORTED_OPERATION, version=contracts/5.7.0)'
+        )
+      );
+
+      expect(mockKVStoreContract.setBulk).toHaveBeenCalledWith(
+        ['key1', 'key2'],
+        ['value1', 'value2']
+      );
+    });
+  });
+
+  describe('get', () => {
+    test('should throw an error if key is empty', async () => {
+      await expect(
+        kvStoreClient.get('0x42d75a16b04a02d1abd7f2386b1c5b567bc7ef71', '')
+      ).rejects.toThrow(ErrorKVStoreEmptyKey);
+    });
+
+    test('should throw an error if address is not valid', async () => {
+      await expect(
+        kvStoreClient.get('invalid_address', 'key1')
+      ).rejects.toThrow(ErrorInvalidAddress);
+    });
+
+    test('should return empty string if both address and key are valid and address does not set any value', async () => {
+      const getSpy = vi.spyOn(kvStoreClient, 'get').mockResolvedValue('');
+      const result = await kvStoreClient.get(
+        '0x42d75a16b04a02d1abd7f2386b1c5b567bc7ef71',
+        'key1'
+      );
+      expect(result).toBe('');
+      expect(getSpy).toHaveBeenCalledWith(
+        '0x42d75a16b04a02d1abd7f2386b1c5b567bc7ef71',
+        'key1'
+      );
+    });
+
+    test('should return empty string if both address and key are valid and address set a value', async () => {
+      mockKVStoreContract.get.mockResolvedValue('value1');
+
+      const getSpy = vi.spyOn(kvStoreClient, 'get');
+      const result = await kvStoreClient.get(
+        '0x42d75a16b04a02d1abd7f2386b1c5b567bc7ef71',
+        'key1'
+      );
+      expect(result).toBe('value1');
+      expect(getSpy).toHaveBeenCalledWith(
+        '0x42d75a16b04a02d1abd7f2386b1c5b567bc7ef71',
+        'key1'
+      );
+      expect(mockKVStoreContract.get).toHaveBeenCalledWith(
+        '0x42d75a16b04a02d1abd7f2386b1c5b567bc7ef71',
+        'key1'
+      );
+    });
+
+    test('should throw an error if a network error occurs', async () => {
+      mockKVStoreContract.get.mockRejectedValue(
+        new Error('could not detect network')
+      );
+
+      await expect(
+        kvStoreClient.get('0x42d75a16b04a02d1abd7f2386b1c5b567bc7ef71', 'key1')
+      ).rejects.toThrow(Error('Failed to get value: could not detect network'));
```

Comment:
I would say network error, it might not be that the network couldn't be detected, it might be a forbidden error or similar